### PR TITLE
Set source branch during merge operation for conflicts dialog

### PR DESCRIPTION
**Pull request by tidy-dev**
## Description

Set source branch during merge operation for conflicts dialog

### Screenshots

Screen Shot 2021-06-08 at 10 29 36 AM

## Release notes
Notes: no-notes

### CODE REVIEW

File: app/src/ui/multi-commit-operation/choose-branch/merge-choose-branch-dialog.tsx [hunk 1]
```diff
@@ -29,11 +30,11 @@ export class MergeChooseBranchDialog extends BaseChooseBranchDialog {
 
     dispatcher.mergeBranch(
       repository,
-      selectedBranch.name,
+      selectedBranch,
       this.mergeStatus,
       operation === MultiCommitOperationKind.Squash
     )
-    this.props.onDismissed()
+    this.props.dispatcher.closePopup(PopupType.MultiCommitOperation)
```
Comment (sergiou87) [inline app/src/ui/multi-commit-operation/choose-branch/merge-choose-branch-dialog.tsx:37]: Why this change? 🤔

**Review by sergiou87 — COMMENTED** [review]
Overall LGTM! Only one question 😳

**Review by sergiou87 — APPROVED** [review]

File: app/src/ui/multi-commit-operation/choose-branch/merge-choose-branch-dialog.tsx (hunk 1 above)
Comment (tidy-dev) [reply app/src/ui/multi-commit-operation/choose-branch/merge-choose-branch-dialog.tsx:37]: It was not necessary, but it should have been that way to start with. A preventative measure from this logic interfering with another features logic.
